moveit_ros/moveit_servo/demos/cpp_interface/demo_pose.cpp

```diff
@@ -0,0 +1,143 @@
+  auto demo_node = std::make_shared<rclcpp::Node>("moveit_servo_demo");
+
+  // Get the servo parameters.
+  std::string param_namespace = "moveit_servo";
+  auto servo_param_listener = std::make_shared<const servo::ParamListener>(demo_node, param_namespace);
+  auto servo_params = servo_param_listener->get_params();
+
+  // The publisher to send trajectory message to the robot controller.
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr trajectory_outgoing_cmd_pub =
+      demo_node->create_publisher<trajectory_msgs::msg::JointTrajectory>(servo_params.command_out_topic,
+                                                                         rclcpp::SystemDefaultsQoS());
+
+  // Create the servo object
+  auto planning_scene_monitor = createPlanningSceneMonitor(demo_node, servo_params);
+  auto servo = Servo(demo_node, servo_param_listener, planning_scene_monitor);
+
+  // Wait for some time, so that the planning scene is loaded in rviz.
+  // This is just for convenience, should not be used for sync in real application.
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  // For syncing pose tracking thread and main thread.
+  std::mutex pose_guard;
+  std::atomic<bool> stop_tracking = false;
+
+  // Set the command type for servo.
+  servo.expectedCommandType(CommandType::POSE);
+
+  // The dynamically updated target pose.
+  PoseCommand target_pose;
+  target_pose.frame_id = servo_params.ee_frame;
+  target_pose.pose = Eigen::Isometry3d::Identity();  // Target pose in ee frame (same as the ee frame pose)
+  // Convert the pose to planning frame.
+  target_pose = servo.toPlanningFrame(target_pose);
+
+  // The pose tracking lambda that will be run in a separate thread.
+  auto pose_tracker = [&]() {
+    KinematicState joint_state;
+    rclcpp::WallRate tracking_rate(1 / servo_params.publish_period);
+    while (!stop_tracking)
+    {
+      {
+        std::lock_guard<std::mutex> pguard(pose_guard);
+        joint_state = servo.getNextJointState(target_pose);
+      }
+      StatusCode status = servo.getStatus();
+      if (status != StatusCode::INVALID)
+        trajectory_outgoing_cmd_pub->publish(composeTrajectoryMessage(servo_params, joint_state));
+
+      tracking_rate.sleep();
+    }
+  };
+
+  // Pose tracking thread will exit upon reaching this pose.
+  Eigen::Isometry3d terminal_pose = target_pose.pose;
+  terminal_pose.translate(Eigen::Vector3d(0.0, 0.0, -0.1));
+
+  std::thread tracker_thread(pose_tracker);
+  tracker_thread.detach();
+
+  const double delta = 0.002;
+  rclcpp::WallRate command_rate(50);
+  RCLCPP_INFO_STREAM(LOGGER, servo.getStatusMessage());
+
+  while (!stop_tracking)
+  {
+    {
+      std::lock_guard<std::mutex> pguard(pose_guard);
+      target_pose.pose = servo.getEndEffectorPose();
+      stop_tracking = target_pose.pose.isApprox(terminal_pose, servo_params.pose_tracking.linear_tolerance);
+      target_pose.pose.translate(Eigen::Vector3d(0.0, 0.0, -delta));
```

Comment:
Is this comment correct? I think it would be nice to highlight dynamic pose tracking.

```suggestion
      // Dynamically update the target pose
      target_pose.pose.translate(Eigen::Vector3d(0.0, 0.0, -delta));
```

Reply:
Yeah, makes it more clear. 👍